staging/rustfmt.toml

```diff
@@ -1,7 +1,76 @@
+hex_literal_case = "Upper"
+match_block_trailing_comma = false
+# Controls how imports are structured in use statements. Imports will be merged
+# or split to the configured level of granularity. (default value is Preserve)
 imports_granularity = "Crate"
+# Unix or Windows line endings (default value is Auto)
+newline_style = "Unix"
+# Convert /* */ comments to // comments where possible. (default value is false)
+normalize_comments = true
+# Convert #![doc] and #[doc] attributes to //! and /// doc comments. (default value is false)
+normalize_doc_attributes = true
+# Reorder impl items. type and const are put first, then macros and methods. (default value is false)
+reorder_impl_items = true
+# Format string literals where necessary. (default value is false)
+format_strings = true
+# Discard existing import groups, and create three groups for:
+#
+#    std, core and alloc,
+#    external crates,
+#    self, super and crate imports.
 group_imports = "StdExternalCrate"
+# Break comments to fit on the line (default value is false)
 wrap_comments = true
+# Minimum number of blank lines which must be put between items. If two items
+# have fewer blank lines between them, additional blank lines are inserted.
+# (default value is 0)
+blank_lines_upper_bound = 1
+# Replace strings of _ wildcards by a single .. in tuple patterns (default value is false)
+condense_wildcard_suffixes = true
+# Use field initialize shorthand if possible. (default value is false)
+use_field_init_shorthand = true
+# Which version of the formatting rules to use. Version::One is
+# backwards-compatible with Rustfmt 1.0. Other versions are only backwards
+# compatible within a major version number. (default value is "One"
+version = "Two"
+
+# Default values
+
+# Max width for code snippets included in doc comments. Only used if
+# format_code_in_doc_comments is true. (default value)
+doc_comment_code_block_width = 100
+# Format the bodies of macros. (default value)
+format_macro_bodies = true
+# Use tab characters for indentation, spaces for alignment. (default value)
+hard_tabs = false
+# Indent style of imports (default value)
+imports_indent = "Block"
+# Item layout inside a imports block. (default value)
+imports_layout = "Mixed"
+# Indent on expressions or items. (default value)
+indent_style = "Block"
+# (default value)
+max_width = 100
+# Merge multiple derives into a single one. (default value)
+merge_derives = true
+# Remove nested parens. (default value)
+remove_nested_parens = true
+# Number of spaces per tab (default value)
+tab_spaces = 4
+# How to handle trailing commas for lists (default value)
+trailing_comma = "Vertical"
+# Add trailing semicolon after break, continue and return. (default value)
+trailing_semicolon = true
+# Reorder import and extern crate statements alphabetically in groups (a group is separated by a newline). (default value)
+reorder_imports = true
+# Reorder mod declarations alphabetically in group. (default value)
+reorder_modules = true
```

Comment:
Do we set these to default values? Any reason why we set them at all then? I think we probably want to follow the overall Rust defaults but only tweak a few options that we find useful?

Reply:
I thought to include them in case the defaults change. They are not unreasonable to have and sometimes rust rolls back on choices. Is it that bad, you think?

Reply:
I am just worried that nobody will ever review the delta that we got compared to Rust default. Then we are left with code laws that only exist because they happened to be default at some point. IMHO saying "we track Rust defaults, expect these few stricter rules" keeps our rule set simpler.

Reply:
Yes you're right. We're not able to easily see changes against current defaults so it can confusing. I'll remove them

Reply:
The convention I tend to see is the defaults are described but commented out.